miio/curtain_youpin.py

```diff
@@ -0,0 +1,290 @@
+        """
+        return NightTipLight(self.data["night_tip_light"])
+
+    @property
+    def run_time(self) -> int:
+        """Run time
+        Value range: [0, 255, 1]
+        """
+        return self.data["run_time"]
+
+    @property
+    def current_position(self) -> int:
+        """Current position
+        Value range: [0, 100, 1]
+        """
+        return self.data["current_position"]
+
+    @property
+    def target_position(self) -> int:
+        """Target position
+        Value range: [0, 100, 1]
+        """
+        return self.data["target_position"]
+
+    @property
+    def adjust_value(self) -> int:
+        """ Adjust value
+        Value range: [-100, 100, 1]
+        """
+        return self.data["adjust_value"]
+
+    def __repr__(self) -> str:
+        s = (
+            "<CurtainStatus"
+            "status=%s,"
+            "polarity=%s,"
+            "position_limit=%s,"
+            "night_tip_light=%s,"
+            "run_time=%s,"
+            "current_position=%s,"
+            "target_position=%s,"
+            "adjust_value=%s>"
+            % (
+                self.status,
+                self.polarity,
+                self.position_limit,
+                self.night_tip_light,
+                self.run_time,
+                self.current_position,
+                self.target_position,
+                self.adjust_value
+            )
+        )
+        return s
+
+class CurtainMiot(MiotDevice):
+    """Main class representing the curtain which uses MIoT protocol."""
+
+    def __init__(
+        self,
+        ip: str = None,
+        token: str = None,
+        start_id: int = 0,
+        debug: int = 0,
+        lazy_discover: bool = True,
+    ) -> None:
+        super().__init__(_MAPPING, ip, token, start_id, debug, lazy_discover)
+
+    @command(
+        default_output=format_output(
+            "",
+            "Device status: {result.status}\n"
+            "Manual enabled: {result.manual_enabled}\n"
+            "Polarity: {result.polarity}\n"
+            "Position limit: {result.position_limit}\n"
+            "Enabled night tip light: {result.night_tip_light}\n"
+            "Run time: {result.run_time}\n"
+            "Current position: {result.current_position}\n"
+            "Target position: {result.target_position}\n"
+            "Adjust value: {result.adjust_value}\n",
+        )
+    )
+    def status(self) -> CurtainStatus:
+        """Retrieve properties."""
+
+        return CurtainStatus(
+            {
+                prop["did"]: prop["value"] if prop["code"] == 0 else None
+                for prop in self.get_properties_for_mapping()
+            }
+        )
+
+    @command(
+        click.argument("motor_control", type=EnumType(MotorControl)),
+        default_output=format_output("Set motor control to {motor_control}"),
+    )
+    def set_motor_control(self, motor_control: MotorControl):
+        """Set motor control.
+        Values: Pause, Open, Close, Auto
```

Comment:
```suggestion
```

Reply:
These are already documented by the enum, so no need for separate listing in docstring.